scene/resources/tile_set.cpp

```diff
@@ -1063,10 +1063,17 @@ void TileSet::move_custom_data_layer(int p_from_index, int p_to_pos) {
 void TileSet::remove_custom_data_layer(int p_index) {
 	ERR_FAIL_INDEX(p_index, custom_data_layers.size());
 	custom_data_layers.remove_at(p_index);
-	for (KeyValue<String, int> E : custom_data_layers_by_name) {
+
+	bool removed = false;
+	for (KeyValue<String, int> &E : custom_data_layers_by_name) {
+		if (removed) {
```

Comment:
```suggestion
		if (E.value > p_index) {
```
Unless something ensures that the map is iterated in order of the values?


Also removing the item like this is an asan crash now, and probably in extremely rare cases without asan as well, so I would move the erase after the loop.